src/Microsoft.AspNet.OData.Shared/Builder/ODataConventionModelBuilder.cs

```diff
@@ -694,41 +698,52 @@ private void MapStructuralProperty(StructuralTypeConfiguration type, PropertyInf
 
         // figures out the type of the property (primitive, complex, navigation) and the corresponding edm type if we have seen this type
         // earlier or the user told us about it.
-        private PropertyKind GetPropertyType(PropertyInfo property, out bool isCollection, out IEdmTypeConfiguration mappedType)
+        private PropertyKind GetPropertyType(PropertyInfo property, out bool isCollection, out IEdmTypeConfiguration mappedType, out bool IsValid)
         {
             Contract.Assert(property != null);
 
+            IsValid = true;
+            mappedType = null;
+            isCollection = false;
+
+            if (typeof(ODataIdContainer) == property.PropertyType)
```

Comment:
should we use isassignablefrom in case the user has derived from ODataIdContainer?

Reply:
I'd love to take it.